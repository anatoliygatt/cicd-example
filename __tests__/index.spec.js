const { ApolloServer } = require("apollo-server-express");
const { createTestClient } = require("apollo-server-testing");
const typeDefs = require("../typeDefs");
const resolvers = require("../resolvers");

describe("queries", () => {
  let server;

  beforeAll(() => {
    server = new ApolloServer({
      typeDefs,
      resolvers
    });
  });

  it("should return viewer details", async () => {
    const { query } = createTestClient(server);
    const res = await query({
      query: `
        query {
          viewer {
            id
            firstName
            lastName
          }
        }
      `
    });
    expect(res.data).toMatchSnapshot();
  });
});
