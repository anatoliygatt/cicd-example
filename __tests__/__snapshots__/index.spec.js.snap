// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries should return viewer details 1`] = `
Object {
  "viewer": Object {
    "firstName": "Anatoliy",
    "id": "5c0e3aad39c4c3481c0b8743",
    "lastName": "Gatt",
  },
}
`;
